ingest.py

```diff
@@ -139,6 +139,86 @@ def does_vectorstore_exist(persist_directory: str) -> bool:
+        """
+        print("Existing directories in ./sources:")
+        directories = os.listdir("./sources")
+        for index, directory in enumerate(directories, start=1):
+            print(f"{index}. {directory}")
+
+    def _create_directory(directory_name):
+        """
+        This function creates a new directory with the given directory_name in the ./sources directory.
+        It also creates a corresponding directory in the ./dbs directory for the database files.
+        It sets the directory paths as environment variables and returns them.
+
+        Parameters:
+        directory_name (str): The name for the new directory.
+
+        Returns:
+        directory_path (str): The path of the new directory.
+        db_path (str): The path of the database directory for the new directory.
+        """
+        directory_path = f"./sources/{directory_name}"
+        db_path = f"./dbs/{directory_name}"
+        os.makedirs(directory_path)
+        os.makedirs(db_path)
+        set_key('.env', 'SOURCE_DIRECTORY', directory_path)
+        set_key('.env', 'PERSIST_DIRECTORY', db_path)
+        print(f"Created new directory: {directory_path}")
+        return directory_path, db_path
+
+    while True:
+        print("Select an option or 'q' to quit:")
+        print("1. Select existing directory")
+        print("2. Create a new directory")
+        choice = input("Enter your choice (1 or 2): ")
+        if choice == "1":
+            _display_directories()
+            existing_directory = input("Enter the number of the existing directory: ")
+            try:
+                selected_directory = os.listdir("./sources")[int(existing_directory) - 1]
+                selected_directory_path = f"./sources/{selected_directory}"
+                selected_db_path = f"./dbs/{selected_directory}"
+                if not os.listdir(selected_directory_path):
+                    print(f"Error: Directory '{selected_directory}' is empty.")
+                    print("Please create files in the directory or choose another.")
+                else:
+                    if not os.path.exists(selected_db_path):
+                        os.makedirs(selected_db_path)
+                    set_key('.env', 'SOURCE_DIRECTORY', selected_directory_path)
+                    set_key('.env', 'PERSIST_DIRECTORY', selected_db_path)
+                    print(f"Selected directory: {selected_directory_path}")
+                    break
+            except (ValueError, IndexError):
+                print("Invalid directory number. Please try again.")
+        elif choice == "2":
+            new_directory_name = input("Enter the name for the new directory: ")
+            selected_directory_path, selected_db_path = _create_directory(new_directory_name)
+            input("Place your source material into the new folder and press enter to continue...")
+            break
+        elif choice == "q":
```

Comment:
Suggestions to add choice option num 3 :
```python

print(f"3. Current source_directory is {source_directory}. Do you want to use it?")

..

elif choice == "3":
  return source_directory, persist_directory

```

Reply:
Even though I would extract this input prompting to separate method and add timeout that option 3 is the default one executed if there's no input:

```python
    def _get_user_choice(timeout):
        print("Select an option or 'q' to quit:")
        print("1. Select an existing directory")
        print("2. Create a new directory")
        print(f"3. Use current source_directory: {source_directory}")
        inputs = [sys.stdin]
        readable, _, _ = select.select(inputs, [], [], timeout)
        if readable:
            return sys.stdin.readline().strip()
        return "3"
```
code around:
```
import sys
import select
...
def prompt_user(timeout=10):
...
choice = _get_user_choice(timeout)
```

Reply:
Good idea, I agree with this approach, added and tested, seems to work well.